PyDynamic/model_estimation/fit_filter.py

```diff
@@ -500,170 +615,140 @@ def invLSFIR_uncMC(H, UH, N, tau, f, Fs, wt=None, verbose=True):
-        a : np.ndarray
-            The IIR filter denominator coefficient vector in a 1-D sequence
-        tau : int
-            time delay (in samples)
+    b : array_like
+        The IIR filter numerator coefficient vector in a 1-D sequence.
+    a : array_like
+        The IIR filter denominator coefficient vector in a 1-D sequence.
+    tau : int
+        Filter time delay (in samples).
 
     References
     ----------
-        * Eichstädt, Elster, Esward, Hessling [Eichst2010]_
+    * Eichstädt, Elster, Esward, Hessling [Eichst2010]_
 
     """
-    from numpy import count_nonzero, roots, ceil, median
-
-    if verbose:
-        print(
-            "\nLeast-squares fit of an order %d digital IIR filter to the" % max(Nb, Na)
-        )
-        print("reciprocal of a frequency response given by %d values.\n" % len(Hvals))
-
-    w = 2 * np.pi * f / Fs
-    Ns = np.arange(0, max(Nb, Na) + 1)[:, np.newaxis]
-    E = np.exp(-1j * np.dot(w[:, np.newaxis], Ns.T))
-
-    bi, ai = _fitIIR(Hvals, tau, w, E, Na, Nb, inv=True)
-
-    if justFit:  # no uncertainty evaluation
-        return bi, ai
-
-    if count_nonzero(abs(roots(ai)) > 1) > 0:
-        stable = False
-    else:
-        stable = True
-
-    maxiter = 50
-
-    astab = mapinside(ai)  # stabilise filter
-    run = 1
-
-    while stable is not True and run < maxiter:  # shift delay such that filter
-        # becomes stable
-        g1 = grpdelay(bi, ai, Fs)[0]
-        g2 = grpdelay(bi, astab, Fs)[0]
-        tau = ceil(tau + median(g2 - g1))
-
-        bi, ai = _fitIIR(Hvals, tau, w, E, Na, Nb, inv=True)
-        if count_nonzero(abs(roots(ai)) > 1) > 0:
-            astab = mapinside(ai)
-        else:
-            stable = True
-        run = run + 1
-
-    if count_nonzero(abs(roots(ai)) > 1) > 0 and verbose:
-        print("Caution: The algorithm did NOT result in a stable IIR filter!")
-        print(
-            "Maybe try again with a higher value of tau0 or a higher filter order?"
-        )
-
-    if verbose:
-        print("Least squares fit finished after %d iterations (tau=%d).\n" % (run, tau))
-        Hd = dsp.freqz(bi, ai, 2 * np.pi * f / Fs)[1]
-        Hd = Hd * np.exp(1j * 2 * np.pi * f / Fs * tau)
-        res = np.hstack((np.real(Hd) - np.real(Hvals), np.imag(Hd) - np.imag(Hvals)))
-        rms = np.sqrt(np.sum(res ** 2) / len(f))
-        print("Final rms error = %e \n\n" % rms)
-
-    return bi, ai, int(tau)
+    warn(
+        "invLSIIR: The method invLSIIR() will be removed in the next major release "
+        "3.0.0. Please switch to the current method LSIIR() with the same input "
+        "parameters as in your call of invLSIIR in the same order and "
+        "additionally specify inv=True.",
+        PendingDeprecationWarning,
+    )
+    return LSIIR(
+        Hvals=Hvals,
+        Nb=Nb,
+        Na=Na,
+        f=f,
+        Fs=Fs,
+        tau=tau,
+        justFit=justFit,
+        verbose=verbose,
+        inv=True,
+    )
```

Comment:
Alternatively we could as well only raise a DeprecationWarning/Exception and not call LSIIR validly. We should discuss this.

Reply:
I would prefer not having (again) a Deprecation Warning for a long time whenever starting PyDynamic. Also in a new release we shouldn't have a deprWarn so early if not really necessary.

Reply:
Resolved in 1646a31 and with #192 .